commit 5a6c3296a7a90694ad4042f6256f3da6d4fa4ee8
Author: James Clarke <jrtc27@jrtc27.com>
Date:   Fri Jan 19 17:22:49 2018 +0000

    gdb: Fix ia64 defining TRAP_HWBKPT before including gdb_wait.h
    
    On ia64, gdb_wait.h eventually includes siginfo-consts-arch.h, which
    contains an enum with TRAP_HWBKPT, along with a #define. Thus we cannot
    define TRAP_HWBKPT to 4 beforehand, and so gdb_wait.h must be included
    earlier; include it from linux-ptrace.h so it can never come afterwards.
    
    gdb/ChangeLog:
    
            * nat/linux-ptrace.c: Remove unnecessary reinclusion of
            gdb_ptrace.h, and move including gdb_wait.h ...
            * nat/linux-ptrace.h: ... to here.

---
 gdb/nat/linux-ptrace.c |    2 --
 gdb/nat/linux-ptrace.h |    1 +
 2 files changed, 1 insertion(+), 2 deletions(-)

--- a/gdb/nat/linux-ptrace.c
+++ b/gdb/nat/linux-ptrace.c
@@ -21,8 +21,6 @@
 #include "linux-procfs.h"
 #include "linux-waitpid.h"
 #include "buffer.h"
-#include "gdb_wait.h"
-#include "gdb_ptrace.h"
 #include <sys/procfs.h>
 
 /* Stores the ptrace options supported by the running kernel.
--- a/gdb/nat/linux-ptrace.h
+++ b/gdb/nat/linux-ptrace.h
@@ -21,6 +21,7 @@
 struct buffer;
 
 #include "nat/gdb_ptrace.h"
+#include "gdb_wait.h"
 
 #ifdef __UCLIBC__
 #if !(defined(__UCLIBC_HAS_MMU__) || defined(__ARCH_HAS_MMU__))
